Update online beamspot to quadruplet-only and rescale track uncertainties

makortel opened the pull request:
This PR makes the following updates to the online beamspot applications making them more robust:
* rescale pixel track parameter uncertainties to match better the reality
  * the uncertainties come from phase0 parametrization, so a difference wrt. phase1 is expected
  * @rovere tuned the scale factor such that the pulls of pixel vertex position coordinates were correct
* do pixel tracks only from quadruplets
  * triplets produce too many fakes that introduce small biases in the beamspot position
* enlargen the tracking region in Z from +- 3 cm to +- 6 cm

Tested in CMSSW_9_2_X_2017-06-09-1100. Should have no impact on runTheMatrix workflows.

@rovere @VinInn @mtosi @JanFSchulte @dinardo @sikler

rovere committed:
Quadruplet only pixel tracks

rovere committed:
Rescale Error for PixelTracks for PhaseI

makortel committed:
Cleanup

Avoid storing ParameterSet and parsing it once per event.

Remove "Quadruplets" from scaleErrorsForPhaseI as the rescaling is
inteded for both quadruplets and triplets when the first hit is on
BPix1.

Remove now-unused triplet configuration from the pixel tracking configuration.

makortel committed:
Enlargen the region in Z from 3 to 6 cm

makortel commented:
@cmsbuild, please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @makortel (Matti Kortelainen) for master.

It involves the following packages:

DQM/Integration
RecoPixelVertexing/PixelTrackFitting


@perrotta, @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@batinkov, @felicepantaleo, @GiacomoSguazzoni, @rovere, @VinInn, @thomreis, @ebrondol, @threus, @dgulhan this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
-1

Tested at: 6fe30c150335eafb0420033bf34326555132e109

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19267/20630/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
136.731 step1
<pre>DAS Error</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

dinardo commented:
Dear Experts,
concerning the error rescale factor, as you might have seen in the BeamPixel Online DQM monitor there is already a vertex error scale factor: the so called "VxErrCorr" in the beampixel_dqm_sourceclient-live_cfg.py configuration file.

Is that number still valid, or given your recent modifications that number should be now "1"?

Many thanks and kind regards,
- Mauro.

slava77 commented:
@cmsbuild please test

it should work this time

cmsbuild commented:
The tests are being triggered in jenkins.

rovere commented:
@dinardo 
Ciao Mauro,
in fact after deploying this PR in the DQM Playback system we had improved results from the beam_dqm module but almost no results from your beampixel module.
Can you test this PR with this year's data and check why with this PR the situation is worst? Maybe the parameter you were referring to could play a role.

Thanks,
M.

cmsbuild commented:
+1
Tested at: 6fe30c150335eafb0420033bf34326555132e109
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19267/20638/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19267/20638/summary.html

Comparison Summary:
 - You potentially added 2 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1803136
 - DQMHistoTests: Total failures: 29341
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1773629
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented on RecoPixelVertexing/PixelTrackFitting/interface/PixelFitterByHelixProjections.h:
```diff
@@ -10,29 +10,21 @@
-private:
   const edm::EventSetup *theES;
   const MagneticField *theField;
+  const bool thescaleErrorsForPhaseI;
```
how does an instance of this class know that it operates on a phase-1 input or a phase-1 environment?
it looks like ```ForPhase1``` can be omitted

slava77 replied:
actually, based on implementation it should probably be
thescaleErrorsForBPIX1

makortel replied:
I can certainly rename the parameter and variable to `theScaleErrorsForPhase1BPix1` or similar (I'd keep "phase1" in the name as the tuning was done with phase1 pixel, and is not necessarily applicable to phase2 (and clarly not for phase0)).

makortel commented on RecoPixelVertexing/PixelTrackFitting/python/pixelFitterByHelixProjections_cfi.py:
```diff
@@ -0,0 +1,8 @@
+
+pixelFitterByHelixProjections = pixelFitterByHelixProjectionsDefault.clone()
+
+phase1Pixel.toModify( pixelFitterByHelixProjections, scaleErrorsForPhaseI = True)
```
@slava77 `PixelFitterByHelixProjections` knows whether it operates on phase1 detector or not from this era customization.

slava77 commented:
On 6/22/17 4:53 AM, Matti Kortelainen wrote:
> *@makortel* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In
> RecoPixelVertexing/PixelTrackFitting/interface/PixelFitterByHelixProjections.h
> <https://github.com/cms-sw/cmssw/pull/19267#discussion_r123489707>:
> 
>>    const edm::EventSetup *theES;
>    const MagneticField *theField;
> +  const bool thescaleErrorsForPhaseI;
> 
> I can certainly rename the parameter and variable to
> |theScaleErrorsForPhase1BPix1| or similar (I'd keep "phase1" in the name
> as the tuning was done with phase1 pixel, and is not necessarily
> applicable to phase2 (and clarly not for phase0)).

If you prefer to keep "phase1" in the name, please add appropriate
software protections
that confirm that this is applied on phase1
(e.g. someone accidentally applying it on phase-0  or phase-2 should get
an exception or other error).

Otherwise, this modifier in the name appears inappropriate to me.
One could just as well desire to apply the same (with perhaps a
different value in the scale)
 for phase-2 for whatever reason and will generate quite a bit of confusion.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19267#discussion_r123489707>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbvOgiUZodMt-4dXnfoi8DOcpfSiQks5sGlXUgaJpZM4N8Hhf>.
>

slava77 commented:
On 6/22/17 4:54 AM, Matti Kortelainen wrote:
> @slava77 <https://github.com/slava77> |PixelFitterByHelixProjections|
> knows whether it operates on phase1 detector or not from this era
> customization.
> 

My point is that if this name is to be kept in the parameter,
it should be impossible to apply it (by mistake or intentionally)  for
any other era/modifier

makortel commented:
@slava77 Right, as the scale factor is also given as a parameter there should be no need to have "phase1" in the prameter/variable name (it is the scale factor that is phase1-specific, not the scaling itself). Will change along you suggested (but I'd prefer to wait until you have completed the review).

slava77 commented:
On 6/22/17 6:29 AM, Matti Kortelainen wrote:
> @slava77 <https://github.com/slava77> Right, as the scale factor is also
> given as a parameter there should be no need to have "phase1" in the
> prameter/variable name (it is the scale factor that is phase1-specific,
> not the scaling itself). Will change along you suggested (but I'd prefer
> to wait until you have completed the review).
> 

I was done with the review yesterday.
Sorry for lack of an explicit message.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/19267#issuecomment-310380338>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbt8T8hzw2LRDff-srqxvY-C-VHhsks5sGmxUgaJpZM4N8Hhf>.
>

makortel committed:
Rename scaleErrorsForPhaseI -> scaleErrorsForBPix1

This is more consistent as the scaling is done for BPix1, whereas it
is the scale factor value that is specific for phase1.

dmitrijus commented:
+1

slava77 commented:
@makortel 
please clarify if you are planning to update this soon
Thank you.

makortel commented:
I'll push an update later today. 

On 23 June 2017 18:31:01 CEST, Slava Krutelyov <notifications@github.com> wrote:
>@makortel 
>please clarify if you are planning to update this soon
>Thank you.

-- 
Sent from my mobile.

cmsbuild commented:
Pull request #19267 was updated. @perrotta, @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

makortel commented:
@cmsbuild, please test

Renamed the parameter/variable as suggested by @slava77.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ebacaf6248b0316ff0ccdbfeb7c55028b79a27f6
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19267/20895/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5ad2ca330590b7d14a64a053f7f4350de7357460
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19267/20895/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19267/20895/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19267/20895/summary.html

Comparison Summary:
 - You potentially added 5 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1802882
 - DQMHistoTests: Total failures: 44287
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1758429
 - DQMHistoTests: Total skipped: 166
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/19267	ebacaf6
- changes needed for online beamspot setup do not affect the standard reco processing
- jenkins tests pass and comparisons with baseline show no differences

slava77 commented:
@makortel 
I guess now this also needs to be back-ported to 92X.

makortel commented:
Backport to 92X is here #19438.

davidlange6 commented:
merge
